### PR TITLE
test: Expand interaction capabilities in test cases using fixtures

### PR DESCRIPTION
**Pull request by vinistevam**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

When working on more complex flows using fixtures there was some limitation on overwriting some actions while the test ran. In order to improve and expand the fixture's capabilities the changes were applied:
1. Included missing pieces of the default fixtures such as `ApprovalController` and remove `PersonalMessageManager` and `TypedMessageManager`.
2. Inclusion of the `withPermissionControllerConnectedToTestDapp` to have the app connected to the test dapp.
3.  Preload an initial state when provided by fixtures in the `createStore` instead of overwriting the store's getState function.

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**E2E**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/64a4361e-c3ce-4ef9-b570-f878f6dee12e

**Issue**

fixes https://github.com/MetaMask/mobile-planning/issues/1217

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: e2e/specs/permission-system-revoke-single-account-fixture.spec.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  afterAll(async () => {
+    await stopFixtureServer();
+  });
+
+  beforeEach(() => {
+    jest.setTimeout(150000);
+  });
+
+  it('should create new wallet', async () => {
```
Comment (cortisiko) [inline e2e/specs/permission-system-revoke-single-account-fixture.spec.js:38]: Small comment. Rename this test.

Comment (vinistevam) [reply e2e/specs/permission-system-revoke-single-account-fixture.spec.js]: ow, I just committed this test to ensure it was working on bitrise https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/64a4361e-c3ce-4ef9-b570-f878f6dee12e . I removed it now and PR is ready for review.

**Review by cortisiko — DISMISSED** [review]
PR is ✅

**Review by Andepande — DISMISSED** [review]

**Review by OGPoyraz — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]

**Pull request merged**